nzgmdb/data_processing/quality_db.py

```diff
@@ -491,6 +493,235 @@ def apply_clipNet_filter(
+    """
+    # Load the sensitivity ignore file from the data registry
+    sensitivity_ignore = pd.read_csv(NZGMDB_DATA.fetch("sensitivity_ignore.csv"))
+
+    # Ensure datetime columns are in datetime format
+    catalogue["datetime"] = pd.to_datetime(catalogue["datetime"])
+    sensitivity_ignore["start_date"] = pd.to_datetime(sensitivity_ignore["start_date"])
+    sensitivity_ignore["end_date"] = pd.to_datetime(sensitivity_ignore["end_date"])
+
+    # Ensure the dtypes are correct for merging
+    for col in ["sta", "chan", "loc"]:
+        catalogue[col] = catalogue[col].astype(str)
+        sensitivity_ignore[col] = sensitivity_ignore[col].astype(str)
+
+    # Merge on sta, chan, loc to find records that have the same sta chan and loc
+    # as the sensitivity ignore records
+    merged = pd.merge(
+        catalogue,
+        sensitivity_ignore,
+        on=["sta", "chan", "loc"],
+        how="inner",
+        suffixes=("", "_ignore"),
+    )
+
+    # Filter where catalogue datetime is within the ignore period
+    sensitivity_filter = merged[
+        (merged["datetime"] >= merged["start_date"])
+        & (merged["datetime"] <= merged["end_date"])
+    ]
+
+    # Remove the bypass records if they exist
+    if bypass_records is not None:
+        sensitivity_filter = sensitivity_filter[
+            ~sensitivity_filter["record_id"].isin(bypass_records)
+        ]
+
+    # Create the skipped_records dataframe from sensitivity_filter
+    skipped_records = pd.DataFrame(
+        {
+            "record_id": sensitivity_filter["record_id"],
+            "reason": "Troublesome sensitivity record",
+        }
+    )
+
+    # Filter out the sensitivity records out of the catalogue
+    catalogue = catalogue[~catalogue["record_id"].isin(sensitivity_filter["record_id"])]
+
+    return catalogue, skipped_records
+
+
+def filter_empirical_predictions(
+    catalogue: pd.DataFrame,
+    bypass_records: np.ndarray = None,
+    mean_residual_threshold: float = None,
+    max_residual_threshold: float = None,
+):
+    """
+    This function checks the difference in empirical estimated values and the results from
+    the catalogue. If the difference exceeds certain thresholds, the record is skipped.
+    Note: the empirical predictions are based on the Atkinson 2022 model for subduction interface and slab,
+    and crustal for any other tectonic type.
+
+    Parameters
+    ----------
+    catalogue : pd.DataFrame
+        The catalogue dataframe to filter
+    bypass_records : np.ndarray, optional
+        The records to bypass the quality checks
+    mean_residual_threshold : float, optional
+        The threshold for the mean residual difference, by default grabs from the config
+    max_residual_threshold : float, optional
+        The threshold for the max residual difference, by default grabs from the config
+
+    Returns
+    -------
+    pd.DataFrame
+        The filtered catalogue
+    pd.DataFrame
+        The skipped records
+    """
```

Comment:
To provide a reference to a scientific work in numpydoc style.


```suggestion
    References
    ----------------
    .. [0] REFERENCE HERE.
    """
```

Reply:
An explicit APA reference would be helpful here because searching for "Atkinson 2022 subduction interface" in google scholar does not show me an obvious paper.